salt/modules/status.py

```diff
@@ -366,8 +366,12 @@ def nproc():
 
         salt '*' status.nproc
     '''
-    data = __salt__['cmd.run']('nproc')
+    cmd = {
+        'Linux': 'nproc',
+        'FreeBSD': 'sysctl -n hw.ncpu',
+    }
     try:
+        data = __salt__['cmd.run'](cmd[__grains__['kernel']])
```

Comment:
@thatch45 I meant, if we run on an OS that's not in the list .. should we error out somehow ? or is this ok ?